src/pixelflow/core.py

```diff
@@ -225,3 +278,78 @@ def pixelflow(
         pf_result.image_intensity = features_img
 
     return pf_result
+
+
+def calc_spacing(
+    coords: tuple[float],
+    mask: npt.NDArray,
+) -> tuple[float]:
+    """Calculate the pixel size for the image in the chosen units.
+
+    Parameters
+    ----------
+    coords : tuple
+        The coordinates of the image in chosen units in the format
+        (top, left, bottom, right) for a 2D image
+    mask : array
+
+    Returns
+    -------
+    spacing : tuple
+        The pixel size in the chosen units
+    """
+
+    # calculate the spacing based on corner coords and number of pixels for each dim
+    spacing = tuple(
+        round(abs((coords[i] - coords[i + mask.ndim]) / mask.shape[i]), 10)
+        for i in range(mask.ndim)
+    )
+
+    # check whether coords are small enough to have problems with rounding to 10dp
```

Comment:
```suggestion
    # check whether coords are small enough to have problems with rounding to 10dp
```
what do you mean by dp?

Reply:
decimal places, happy to spell it out